packages/cli/src/util/paths.ts

```diff
@@ -31,9 +31,28 @@ export function createExamplePath(
 ): E.Either<Error, string> {
   return DoEither.bind('pathData', generateTemplateAndValuesForPathParams(operation))
     .bindL('queryData', ({ pathData }) => generateTemplateAndValuesForQueryParams(pathData.template, operation))
-    .return(({ pathData, queryData }) =>
-      URI.expand(queryData.template, transformValues({ ...pathData.values, ...queryData.values }))
-    );
+    .return(({ pathData, queryData }) => {
+      // replace "-" in template path and query params
+      const cleanedTemplate = queryData.template.replace(/(?<=\{.*)(?=.*\})(-)/g, '');
+      const realValues = transformValues({ ...pathData.values, ...queryData.values });
+      const cleanedValues = {};
+      for (const realValue in realValues) {
+        const cleanedValue = realValue.replace(/-/g, '');
+        cleanedValues[cleanedValue] = realValues[realValue];
+      }
+
+      const cleanedExpandedPath = URI.expand(cleanedTemplate, cleanedValues);
+
+      // replace cleaned values with real values
+      // i.e put "-"" back in
+      let finalExpandedPath = cleanedExpandedPath;
+      for (const realValue in realValues) {
+        const cleanedValue = realValue.replace(/-/g, '');
+        finalExpandedPath = cleanedExpandedPath.replace(cleanedValue, realValue);
+      }
```

Comment:
@XVincentX this is not a very good way to get the real query/path param names back into the expanded url. For example, if a query param with hyphen removed exists in other parts of the path the wrong part of the string will be replaced.

Example: 
- path of `/path-path`
- query param of `p-a`
- `cleanedExpandedPath` will be `/path-path?pa=test`
- the query param with hyphen removed is `pa`
- you'll end up doing `/path-path?pa=test`.replace('pa', 'p-a')` which will replace the wrong part of the path and not actually replace the query param

Any suggestions for a more targeted, safer replacement of the path/query params in the `cleanedExpandedPath`?

Reply:
I added a more targeted approach in 6fd670e that uses `urijs` package to do a more targeted replace. This fixes the issue but introduces another issue. It auto-encodes so we're now getting `+` instead of `%20`. 

I also added 2 tests in 28cba6a that will fail if we stick with the original unsafe replace approach to illustrate the issue further and test different approaches

Reply:
urijs has a function to not escape `%20` to `+` called `escapeQuerySpace(false)`. This wasn't available to use until I added `"@types/urijs": "^1.19.8"` as a dev dependency. However this devDependency is messing with another packages `uri` namespace, causing the build to fail. Looking into ways to fix this.